app/code/Magento/CatalogCustomerGraphQl/Model/Resolver/PriceTiers.php

```diff
@@ -120,42 +120,89 @@ function () use ($productId, $context) {
 
                 $productPrice = $this->tiers->getProductRegularPrice($productId) ?? 0.0;
                 $tierPrices = $this->tiers->getProductTierPrices($productId) ?? [];
-
-                return $this->formatProductTierPrices($tierPrices, $productPrice, $store);
+                return $this->filterAndFormatProductTierPrices($tierPrices, $productPrice, $store);
             }
         );
     }
 
     /**
-     * Format tier prices for output
+     * Filter and format tier prices for output
      *
      * @param ProductTierPriceInterface[] $tierPrices
      * @param float $productPrice
      * @param StoreInterface $store
      * @return array
      */
-    private function formatProductTierPrices(array $tierPrices, float $productPrice, StoreInterface $store): array
-    {
+    private function filterAndFormatProductTierPrices(
```

Comment:
An idea...

I would rename this to filterAndFormatTierPrices. Having "Product" in the name seems like superfluous information.

In this case, sounds irrelevant if the objects/data that we filter and format are a property of a product or not.

Reply:
**Done**